test/index.js

```diff
@@ -19,19 +19,21 @@ tape('basic nesting', function(t) {
 })
 
 tape('nesting', function(t) {
-  var result = [['foo', ['bar']], ['bar', ['echo c', 'echo d']], ['bat', [['pipe', ['echo e', 'echo f']], 'echo g']], ['baz', ['echo a b c d']]]
+  var result = [ [ 'foo', [ 'bar' ] ], [ 'bar', [ [ 'echo', [ 'c' ] ], [ 'echo', [ 'd' ] ] ] ], [ 'bat', [ [ 'pipe', [ ['echo', ['e'] ], [ 'echo', ['f'] ] ] ], [ 'echo', [ 'g' ] ] ] ], [ 'baz', [ 'echo a b c d' ] ] ]
   t.same(hackfile(read('nesting-basic')), result, 'parses working nested file')
   t.end()
 })
 
 tape('indent dedent', function(t) {
-  var result = [['pipeline', ['foo', ['pipe', ['echo hello', 'transform', 'cat']], ['run', ['echo', 'hello']]]]]
+  var result = [ [ 'pipeline', [ 'foo', [ 'pipe', [ ['echo', ['hello'] ], 'transform', 'cat' ] ], [ 'run', [ 'echo', 'hello' ] ] ] ] ]
   t.same(hackfile(read('indent-dedent')), result, 'parses working nested file with varied indents and dedents')
   t.end()
 })
 
 tape('advanced', function(t) {
-  var result = [["pipeline",["foo", "run echo hi",["map",["curl http","a","b","c",["pipe",["echo hello","transform","cat"]]]],["run",["echo","hi"]]]],["pipeline",["bar","run echo hi"]],["run",["echo bye"]]]
+  var result = [ ['pipeline',['foo',['run',['echo hi']],['map',[['curl',['http']],'a','b','c',['pipe',[['echo',['hello']],'transform','cat']]]],['run',['echo','hi']]]],
+    ['pipeline',['bar',['run',['echo hi']]]],
```

Comment:
yes, I have been treating one-line commands differently.

For example:
```
run echo hi
  echo hi
  echo hi

run
  echo hi
  echo hi
  echo hi

run echo hi
```

output (stringified):
```
[ '["run",[["echo",["hi"]],["echo",["hi"]],["echo",["hi"]]]]',
  '["run",[["echo",["hi"]],["echo",["hi"]],["echo",["hi"]]]]',
  '["run",["echo hi"]]' ]
```

But I can easily change it so that the output is:
```
[ '["run",[["echo",["hi"]],["echo",["hi"]],["echo",["hi"]]]]',
  '["run",[["echo",["hi"]],["echo",["hi"]],["echo",["hi"]]]]',
  '["run",["echo",  ["hi"]]' ]
```
(so the one-line command is treated more like the other nested commands now)

I'm a fan of this.  What do you think?